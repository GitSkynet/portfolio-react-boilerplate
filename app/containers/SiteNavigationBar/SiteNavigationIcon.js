import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';

const SiteNavigationIcon = styled(FontAwesome).attrs({
  name: (props) => props.name || 'meh-o',
  fixedWidth: (props) => props.fixedWidth,
})`
  display: none !important;

  @media (min-width: 600px) {
    display: inherit !important;
    transform: rotate(-90deg);
    margin: 0;
  }
`;

export default SiteNavigationIcon;
