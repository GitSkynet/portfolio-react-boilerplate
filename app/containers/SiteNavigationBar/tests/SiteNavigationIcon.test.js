import React from 'react';
import { shallow, mount } from 'enzyme';

import SiteNavigationIcon from '../SiteNavigationIcon';

describe('<SiteNavigationIcon />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<SiteNavigationIcon />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should adopt the "name" prop', () => {
    const name = 'check';
    const renderedComponent = mount(<SiteNavigationIcon name={name} />);
    expect(renderedComponent.prop('name')).toBe(name);
  });

  it('should adopt the "fixedWidth" prop', () => {
    const fixedWidth = true;
    const renderedComponent = mount(<SiteNavigationIcon name={'check'} fixedWidth={fixedWidth} />);
    expect(renderedComponent.prop('fixedWidth')).toBe(fixedWidth);
  });
});
