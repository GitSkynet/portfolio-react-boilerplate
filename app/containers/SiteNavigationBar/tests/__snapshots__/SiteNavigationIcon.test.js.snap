// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteNavigationIcon /> should render and match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <FontAwesome
    className="SiteNavigationIcon__SiteNavigationIcon-cszPoD hfGaUp"
    fixedWidth={undefined}
    name="meh-o"
/>,
  "nodes": Array [
    <FontAwesome
      className="SiteNavigationIcon__SiteNavigationIcon-cszPoD hfGaUp"
      fixedWidth={undefined}
      name="meh-o"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SiteNavigationIcon__SiteNavigationIcon />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StyledComponent {
        "_reactInternalInstance": [Circular],
        "attrs": Object {
          "fixedWidth": undefined,
          "name": "meh-o",
        },
        "context": Object {
          "__styled-components-stylesheet__": undefined,
          "__styled-components__": undefined,
        },
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "generatedClassName": "hfGaUp",
          "theme": Object {},
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <FontAwesome
          className="SiteNavigationIcon__SiteNavigationIcon-cszPoD hfGaUp"
          fixedWidth={undefined}
          name="meh-o"
/>,
        "_debugID": 2,
        "_renderedOutput": <FontAwesome
          className="SiteNavigationIcon__SiteNavigationIcon-cszPoD hfGaUp"
          fixedWidth={undefined}
          name="meh-o"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SiteNavigationIcon__SiteNavigationIcon />,
}
`;
